### PR TITLE
test(typeahead): improve reliability

### PR DESCRIPTION
**Pull request by divdavem**
Before submitting a pull request, please make sure you have at least performed the following:

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/CONTRIBUTING.md) and [DEVELOPER.md](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/DEVELOPER.md) guide.
 - [x] built and tested the changes locally.
 - [x] added/updated any applicable tests.
 - [x] added/updated any applicable API documentation.
 - [x] added/updated any applicable demos.

### CODE REVIEW

**Comment by divdavem** [comment]
Note that without this PR, the tests regularly fail with the following error:

```
[1] Failures:
[1] 1) Typeahead should preserve value previously selected with mouse when reopening with focus then closing without selection
[1]   Message:
[1]     Expected 6 to be 2, 'Wrong number of suggestions'.
[1]   Stack:
[1]     Error: Expected 6 to be 2, 'Wrong number of suggestions'.
[1]         at <Jasmine>
[1]         at /home/runner/work/ng-bootstrap/ng-bootstrap/e2e-app/src/app/typeahead/focus/typeahead-focus.pw-spec.ts:11:26
[1]         at <Jasmine>
[1]         at fulfilled (/home/runner/work/ng-bootstrap/ng-bootstrap/node_modules/tslib/tslib.js:111:62)
```
(cf https://github.com/divdavem/ng-bootstrap/runs/1905130301?check_suite_focus=true, running 4 times successfully in 10min before this error happens)

Or this one which is very similar:

```
[1] Failures:
[1] 1) Typeahead should preserve value previously selected with mouse when reopening with focus then closing without selection
[1]   Message:
[1]     Expected 10 to be 2, 'Wrong number of suggestions'.
[1]   Stack:
[1]     Error: Expected 10 to be 2, 'Wrong number of suggestions'.
[1]         at <Jasmine>
[1]         at /home/runner/work/ng-bootstrap/ng-bootstrap/e2e-app/src/app/typeahead/focus/typeahead-focus.pw-spec.ts:11:26
[1]         at <Jasmine>
[1]         at fulfilled (/home/runner/work/ng-bootstrap/ng-bootstrap/node_modules/tslib/tslib.js:111:62)
[1]         at <Jasmine>
[1]         at runNextTicks (internal/process/task_queues.js:62:5)
        at processImmediate (internal/timers.js:434:9)
```
(cf https://github.com/divdavem/ng-bootstrap/runs/1905142666?check_suite_focus=true, running 5 times successfully in 12min before this error happens)

With this PR, e2e playwright tests on webkit are much more reliable:
- https://github.com/divdavem/ng-bootstrap/runs/1912358477?check_suite_focus=true (running 187 times successfully in 6 hours before being interrupted by GitHub)
- https://github.com/divdavem/ng-bootstrap/runs/1912386436?check_suite_focus=true (running 204 times successfully in 6 hours before being interrupted by GitHub)
- https://github.com/divdavem/ng-bootstrap/runs/1912434694?check_suite_focus=true (running 79 times successfully in 2h15min before a test error happens)

File: e2e-app/src/app/typeahead/focus/typeahead-focus.pw-spec.ts [hunk 1]
```diff
@@ -1,57 +1,64 @@
...
 
-  const expectDropdownOpen = async(suggestions = 10) => {
-    const items = await test.page.$$(SELECTOR_TYPEAHEAD_ITEMS);
-    expect(items.length).toBe(suggestions, `Wrong number of suggestions`);
+  const waitForDropdownOpen = async(suggestions = 10) => {
+    await timeoutMessage(
+        test.page.waitForFunction(
+            (args) => (document.querySelectorAll(args[0]).length === args[1]),
+            [SELECTOR_TYPEAHEAD_ITEMS, suggestions] as[string, number]),
+        `Wrong number of suggestions (expected ${suggestions})`);
```
Comment (fbasso) [inline e2e-app/src/app/typeahead/focus/typeahead-focus.pw-spec.ts:14]: I think you can 'lighten' the code by using a string function, i.e.:
```
test.page.waitForFunction(`document.querySelectorAll(${SELECTOR_TYPEAHEAD_ITEMS}).length === ${number}`)
```

What do you think ?

Comment (divdavem) [reply e2e-app/src/app/typeahead/focus/typeahead-focus.pw-spec.ts:14]: Indeed

**Comment by fbasso** [comment]
LGTM !

**Comment by divdavem** [comment]
@fbasso Thank you for your review!

**Pull request merged**